src/test/ui/lint/issue-54099-camel-case-underscore-types.rs

```diff
@@ -0,0 +1,14 @@
+// compile-pass
+
+#![forbid(non_camel_case_types)]
+#![allow(dead_code)]
+
+// None of the following types should generate a warning
+struct _X {}
+struct __X {}
+struct __ {}
+struct X_ {}
+struct X__ {}
+struct X___ {}
```

Comment:
looking around, I couldn't find many tests for this lint really...

Reply:
e.g., to see whether `X_X` would fail

Reply:
Hmm I actually don't know the answer to that myself -- that is whether `X_X` is camel case or not. Btw there are a few more tests about this in `lint-non-camel-case-types.rs`.